### DISCUSSION TITLE
「手把手教你用 Milvus 和 Towhee 搭建一个 AI 聊天机器人！」搭建这个东西是解决什么问题？

### DISCUSSION THREAD

**Discussion by ponponon** [milvus-io/milvus#27995] [Q&A and General discussion]
[手把手教你用 Milvus 和 Towhee 搭建一个 AI 聊天机器人！](https://zhuanlan.zhihu.com/p/658690305) 整篇文章看下来，发现 Milvus 根本没起作用。在 「AI 聊天机器人」引入 milvus 还是解决什么问题，如果没有向量数据库，又会怎么样？

**Comment by yhmo** [comment]
首先，大模型都是基于某个时间之前的知识搭建的，它本身没有知识更新能力
第二，chatgpt这种大模型允许你输入一些提示语，可以让它组织出更准确更丰富的答案
第三，我想让它根据最新的资讯回答问题，怎么做？客户端来了个问题“2023年10月31日发生了什么国际大事？”，它肯定会胡诌一通。那如何让它不要胡诌？
第四，最简单的做法是，我把2023年10月31日的时事新闻拉下来，分割成很多小知识点(文本段落)，转为向量。问题来时，先搜出相关知识点，把这些知识点喂给chatgpt，让它组织出答案。


聊天机器人的原理也差不多，其实就是把chatgpt不知道的东西存在向量数据库里，然后找出来喂给它组织出让人满意的答案。如果没有这些，它也能回答，只不过没人能阻止它瞎编。

**Comment by yhmo** [comment]
文章里也说了milvus的作用：
ops.ann_insert.osschat_milvus(host=MILVUS_HOST, port=MILVUS_PORT)) -------------------- 这是把文本转换的向量数据存入milvus

ops.ann_search.osschat_milvus(host=MILVUS_HOST, port=MILVUS_PORT, 
                              **{'metric_type': 'IP', 'limit': 3, 'output_fields': ['text']}))  -------------------------- 这是从milvus中搜索出相关文本
